src/styles/withStyles.js

```diff
@@ -42,7 +42,9 @@ const withStyles = styleSheet => BaseComponent => {
               ].join('\n'),
             );
 
-            acc[key] = `${renderedClasses[key]} ${classesProp[key]}`;
+            if (classesProp[key] !== undefined) {
```

Comment:
There isn't a warning for `undefined` value passed for a valid class, so we could either warn, throw, or omit.